share/spack/gitlab/cloud_pipelines/stacks/gcp-base/spack.yaml

```
@@ -0,0 +1,20 @@
+spack:
+  view: false
+  definitions:
+  - arch:
+    - %gcc
+
+  - packages:
+    - stream
+
+  - targets:
+    - target=x86_64_v3
+
+  specs:
+  - matrix:
+    - - $packages
+    - - $arch
+    - - $targets
+
```

You also need to place the ci config here to make sure only gcp runners are picked up.
```suggestion
  ci:
    pipeline-gen:
    - build-jobs:
        tags: [gcp]
```

---

This would also needs an image if you are using one, it should be the same image used in the `.gitlab-ci.yml` config